pgmpy/factors/CPD.py

```diff
@@ -110,9 +110,12 @@ def __init__(self, variable, variable_card, values,
                                     " of strings.")
             self.evidence = evidence
             variables.extend(evidence[::-1])
-            if not len(evidence_card) == len(evidence):
+            if len(evidence_card) < len(evidence):
```

Comment:
But in the previous behavior, the same error message - 
"Cardinality of all evidences not specified"
 would be given even if there are suppose, 4 evidences and we are giving 5 elements in the cardinality.
It wouldn't make much of a difference but the error message would be logically incorrect.